python/cudf/cudf/core/reshape.py

```diff
@@ -369,8 +368,7 @@ def length_check(obj, name):
     if len(columns) == 0:
         return df.select_dtypes(exclude=encode_fallback_dtypes)
     else:
-        df_list = []
-
+        result_df = df.copy()
```

Comment:
Instead of deep copying all of the columns is there any way to deep copy only the columns that aren't overwritten in the loop below?

Reply:
Yes, copying only `columns` which is what will be dropped later in the end before return.

Reply:
Since we are anyway having `columns` preserved in `df`, storing the copy of `df` with dropped `columns` in `result_df`.